scripts/apps/archive/controllers/UploadController.ts

```diff
@@ -245,70 +246,122 @@ export function UploadController(
+                        };
+                    } else {
+                        return {error: {filename: file.name}};
+                    }
+                });
+            } else if (file.type.startsWith('video')) {
+                if (!$scope.allowVideo) {
+                    return Promise.resolve({error: {isAllowedFileType: false}});
+                }
+                return Promise.resolve({
+                    file: file,
+                    getThumbnail: () => Promise.resolve('<i class="icon--2x icon-video"></i>'),
+                });
+            } else if (file.type.startsWith('audio')) {
+                if (!$scope.allowAudio) {
+                    return Promise.resolve({error: {isAllowedFileType: false}});
                 }
-            } else if (/^audio/.test(file.type)) {
-                if ($scope.allowAudio) {
+                return Promise.resolve({
+                    file: file,
+                    getThumbnail: () => Promise.resolve('<i class="icon--2x icon-audio"></i>'),
+                });
+            }
+        })).then((result) => {
+            let uploadOfDisallowedFileTypesAttempted: boolean = false;
+
+            result.forEach((file) => {
+                if (!file.error) {
                     acceptedFiles.push({
-                        file: file,
-                        getThumbnail: () => Promise.resolve('<i class="icon--2x icon-audio"></i>'),
+                        file: file.file,
+                        getThumbnail: file.getThumbnail,
                     });
-                } else {
+                } else if (file.error.isAllowedFileType === false) {
                     uploadOfDisallowedFileTypesAttempted = true;
+                } else {
+                    invalidFiles.push(file.error.filename);
                 }
-            } else {
-                uploadOfDisallowedFileTypesAttempted = true;
-            }
-        });
+            });
 
-        if (uploadOfDisallowedFileTypesAttempted) {
-            const message = gettext('Only the following files are allowed: ')
-                + ($scope.allowPicture ? gettext('image') : '')
-                + ($scope.allowVideo ? ', ' + gettext('video') : '')
-                + ($scope.allowAudio ? ', ' + gettext('audio') : '');
+            if (uploadOfDisallowedFileTypesAttempted) {
+                const message = gettext('Only the following files are allowed: ')
+                    + ($scope.allowPicture ? gettext('image') : '')
+                    + ($scope.allowVideo ? ', ' + gettext('video') : '')
+                    + ($scope.allowAudio ? ', ' + gettext('audio') : '');
 
-            notify.error(message);
-        }
+                notify.error(message);
+            }
+
+            if (invalidFiles.length > 0) {
+                notify.error(
+                    gettext(
+                        `The image you\'re trying to add is smaller than
```

Comment:
How do you know if `invalidFiles` contains images and not other file types? You should either call the variable `invalidImages` or check the type of invalid files.

Can you include the dimensions of the current image to the error message too? `Minimal allowed image size is {{minWidth}}x{{minHeight}}. The size of {{file.name}} is {{file.width}}x{{file.height}}.`

I'd say notifications aren't good for this, because if you are uploading 10 images, the notifications boxes will close sooner than users will be able to read the messages. Can you use a modal with all the messages in one place instead?